File: app/src/main/java/com/ditclear/paonet/lib/network/NetProvider.java
package com.ditclear.paonet.lib.network;

import okhttp3.CookieJar;
import okhttp3.Interceptor;
import okhttp3.OkHttpClient;

/**
 * 页面描述：网络配置
 * <p>
 * Created by ditclear on 2017/7/28.
 */

public interface NetProvider {

    Interceptor[] configInterceptors();

    void configHttps(OkHttpClient.Builder builder);

    CookieJar configCookie();

    RequestHandler configHandler();

    long configConnectTimeoutSecs();

    long configReadTimeoutSecs();

    long configWriteTimeoutSecs();

    boolean configLogEnable();

}

File: app/src/main/java/com/ditclear/paonet/helper/network/NetProvider.kt
package com.ditclear.paonet.helper.network

import okhttp3.CookieJar
import okhttp3.Interceptor
import okhttp3.OkHttpClient

/**
 * 页面描述：网络配置
 *
 *
 * Created by ditclear on 2017/7/28.
 */

interface NetProvider {

    fun configInterceptors(): Array<Interceptor>?

    fun configHttps(builder: OkHttpClient.Builder)

    fun configCookie(): CookieJar?

    fun configHandler(): RequestHandler

    fun configConnectTimeoutSecs(): Long

    fun configReadTimeoutSecs(): Long

    fun configWriteTimeoutSecs(): Long

    fun configLogEnable(): Boolean

}
